### PR TITLE
fixes to "Writing a CF Service Broker" docs found while writing broker sample app (take 2)

### PR DESCRIPTION
**Pull request by stupakov**
fixed the errors and missing info in the Authentication, Catalog Management, and Adding Broker sections.

### CODE REVIEW

File: source/docs/running/architecture/services/writing-service.html.md.erb [hunk 1]
```diff
@@ -65,9 +65,9 @@ Cloud Controller (final release v145+) authenticates with the Broker using HTTP
...
 
-When Cloud Controller fetches a catalog from a broker, it will compare the `unique_id` for services and plans with what is in the  Cloud Controller database. If a service or plan in the broker catalog has a `unique_id` which is not present in the database, a new record will be added to the database. If services or plans in the database are found with `unique_id`s that match the broker catalog, Cloud Controller will update update the records to match the broker’s catalog.
+When Cloud Controller fetches a catalog from a broker, it will compare the `id` of the services and plans with what is in the Cloud Controller database. If a service or plan in the broker catalog has an `id` which is not present in the database, a new record will be added to the database. When adding a broker, if services or plans in the database are found with `id`s that match the broker catalog, the operation will fail. When updating a broker, if services or plans in the database are found with `id`s that match the broker catalog, Cloud Controller will update the matching records.
```
Comment (cf-services) [inline source/docs/running/architecture/services/writing-service.html.md.erb]: Cloud Controller compares id for service and plan in the response with value of unique_id in service and plan table.

File: source/docs/running/architecture/services/writing-service.html.md.erb [hunk 2]
```diff
@@ -105,7 +105,7 @@ If the database has plans which are not found in the broker catalog, and there a
 <tr>
   <td>&nbsp;&nbsp;&nbsp;bindable*</td>
   <td>boolean</td>
-  <td>Whether the service can be bound to applications. Defaults to true.</td>
+  <td>Whether the service can be bound to applications.</td>
```
Comment (cf-services) [inline source/docs/running/architecture/services/writing-service.html.md.erb:108]: If bindable is not provided, is the catalog invalid or does it default to true?

File: source/docs/running/architecture/services/writing-service.html.md.erb [hunk 3]
```diff
@@ -174,25 +177,27 @@ If the database has plans which are not found in the broker catalog, and there a
...
+
+Before executing these two steps, make sure you are authenticated as a cloud controller admin (via the `cf login` command).
 
-#### Adding a Broker to Cloud Foundry
+#### Registering a Broker with Cloud Foundry
 
-Registering or updating a broker causes cloud controller to fetch and validate the catalog from your broker, and update the database with any changes found in the catalog. Your service broker should validate the username and password, otherwise anyone could curl your broker to delete all service instances.
+Registering or updating a broker causes cloud controller to fetch and validate the catalog from your broker, and update the database with any changes found in the catalog.  Registering or updating causes the cloud controller to store the basic auth username and password, which the cloud controller  will send to the broker with every request. Your service broker should validate the username and password sent in every request, otherwise anyone could curl your broker to delete all service instances.
```
Comment (cf-services) [inline source/docs/running/architecture/services/writing-service.html.md.erb]: There's a few superflous spaces in this line.

File: source/docs/running/architecture/services/writing-service.html.md.erb (hunk 2 above)
Comment (cf-services) [inline source/docs/running/architecture/services/writing-service.html.md.erb:108]: verify that docs say bindable is required field

**Comment by kimahoff** [comment]
Thanks, Alex! This is merged.

**Pull request closed without merging**